include/revng/Pipes/LiftPipe.h

```diff
@@ -39,7 +39,9 @@ class LiftPipe {
 
   void run(pipeline::Context &Ctx,
            const FileContainer &SourceBinary,
-           pipeline::LLVMContainer &TargetsList);
+           pipeline::LLVMContainer &Output);
+
+  llvm::Error precondition(const pipeline::Context &Ctx) const;
```

Comment:
Shouldn't we call this `checkPrecondition`?

Reply:
i called it precondition so that the writer of a pipe must write a  function name precondition, which i think in that context

Reply:
I think you didn't finish the sentence.

Reply:
in that context is more explanatory than checkPipe

Reply:
A method called `precondition` makes me think that's the same as `getPrecondition`, not that you're checking it.

Reply:
mhh yeah but what is it a precondition if not a function from something to bool? check precondition would be itself a function from something to bool, and do the exact same as the precondition.

Reply:
It is a function from something to `bool`, but here we're not returning a function, we're running it.
If this was to return a function `something -> bool`, the name would be great, but here we're *running it*, not returning it.

Reply:
yeah but it is the syntax Pipe::precondition that returns the precondition, Pipe::precondition() is the syntax that actually checks it.

Reply:
By this logic the function `Instruction::dump` should be called `Instruction::dumper`.

Reply:
the difference is that dump is a verb first and you can extract a name (dumper). While here we have a name first and we need to extract a verb. a more fitting example is a java interface called ConditionChecker that has a single function from void to bool. That is not a ConditionChecker, it is a Condition, because there is nothing to remove to extract a Condition from a ConditionChecker.

Furthermore this name refers to the the well known concept of preconditions and post conditions, which should hint to the user "this is run first and must be true."

Reply:
I think nouns make sense for fields, using a noun for a method is quite confusing.
Sometimes one elides `is`, or `has` prefixes (still sad), but eliding `check` is a bit too much to me.
But I guess we're overdiscussing this, proceed as you think it's more fit.

Reply:
ok, i will had check